Assets/SEE/UI/Window/MetricWindow.cs

```diff
@@ -0,0 +1,215 @@
+        /// </summary>
+        public GraphElement GraphElement;
+
+        /// <summary>
+        /// Prefab for the <see cref="MetricWindow"/>.
+        /// </summary>
+        private static string WindowPrefab => UIPrefabFolder + "MetricWindow";
+
+        /// <summary>
+        /// Prefab for the <see cref="MetricRowLine"/>.
+        /// </summary>
+        private static string ItemPrefab => UIPrefabFolder + "MetricRowLine";
+
+        protected override void StartDesktop()
+        {
+            base.StartDesktop();
+            CreateUIInstance();
+        }
+
+        /// <summary>
+        /// Searches for a specified string within <paramref name="elements"/>.
+        /// Activates or deactivates them based on the search results.
+        /// GameObjects are considered "active" when the search string has been found within their names.
+        /// </summary>
+        /// <param name="searchQuery"> string to search for </param>
+        /// <param name="searchableObjects"> Dictionary of GameObjects to search in </param>
+        private void InputSearchField(string searchQuery, Dictionary<string, GameObject> searchableObjects)
+        {
+            // Remove Whitespaces
+            searchQuery = searchQuery.Trim();
+            if (searchQuery == null || searchQuery.Trim().Length == 0)
+            {
+                foreach (var ele in searchableObjects)
+                {
+                    ele.Value.SetActive(true);
+                }
+            }
+            else
+            {
+                // Deactivate every GameObject
+                IEnumerable<string> searchList = Search(searchQuery, searchableObjects.Values.ToArray());
+                foreach(var ele in searchableObjects)
+                {
+                    ele.Value.SetActive(false);
+                }
+                // Activate GameObjects that match the search results from the fuzzy search.
+                foreach (string ele in searchList)
+                {
+                    searchableObjects.TryGetValue(ele, out GameObject activeObject);
+                    activeObject.SetActive(true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Returns the attributes of all <paramref name="gameObjects"/> whose name match the
+        /// <paramref name="query"/> using a fuzzy search.
+        /// </summary>
+        /// <param name="query"> the search query (part of an attribute name)</param>
+        /// <param name="gameObjects"> the game objects to search through</param>
+        /// <param name="cutoff"> representing the cutoff score for relevance </param>
+        /// <returns> the attributes whose names match the <paramref name="query"/> </returns>
+        public IEnumerable<string> Search(string query, GameObject[] gameObjects, int limit = 15, int cutoff = 40)
+        {
+            List<string> attributesList = new();
+            foreach (GameObject obj in gameObjects)
```

Comment:
The parameter name `gameObjects` was unnecessarily too general, suggesting that nodes of a graph are to be searched. In fact, this parameter represents the rows of the metrics of a node. Its length is the number of metrics a node may have. So it may be acceptable not to cache it. Calculating it for each search instead of caching it, may be handy in case a metric was added (although that is likely happening very rarely).

I renamed the parameter to `metricRows` to avoid the confusion.